analyzers/src/SonarAnalyzer.Common/Extensions/CompilationExtensions.cs

```diff
@@ -33,6 +33,11 @@ internal static class CompilationExtensions
         // See https://github.com/dotnet/roslyn/issues/3798
         compilation.ObjectType.ContainingAssembly.Name == "mscorlib";
 
+    public static bool ReferencesAny(this Compilation compilation, params KnownAssembly[] assemblies)
```

Comment:
This shouldn't be used like this. What you should do instead is:
```cs
context.RegisterCompilationStartAction(compilationStartContext =>
    {
        var templateArguments = new List<Func<InvocationExpression, SemanticModel>>();
        if (compilationStartContext.Compilation.References(KnownAssembly.Microsoft_Extensions_Logging))
        {
             templateArguments.Add(static (invocation, model) => TemplateArgument(invocation, model, KnownType.Microsoft_Extensions_Logging_LoggerExtensions, MicrosoftExtensionsLogging, "message"));
        }
        // ...
        if (templateArguments.Any())
        {
            compilationStartContext.RegisterSyntaxNodeAction(nodeContext => {  } ); // Use templateArguments here
        }
    });

Reply:
That's just the declaration, he did not add a usage yet. I'm also [waiting for an example](https://github.com/SonarSource/sonar-dotnet/pull/8863#pullrequestreview-1914813171).